### PR TITLE
Feature/always recover all consumers

### PR DESCRIPTION
**Pull request by pierresetteskog**
## Proposed Changes
The client dont recover consumers (topology) if connection is lost during reconnection. Following exceptions is found when tested and known to occure. There should not be any problem to merge in to master to.

## Types of Changes

What types of changes does your code introduce to this project?
_Put an `x` in the boxes that apply_

- [x ] Bug fix (non-breaking change which fixes issue #1047)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause an observable behavior change in existing systems)
- [ ] Documentation improvements (corrections, new content, etc)
- [ ] Cosmetic change (whitespace, formatting, etc)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating
the PR. If you're unsure about any of them, don't hesitate to ask on the
mailing list. We're here to help! This is simply a reminder of what we are
going to look for before merging your code._

- [ x] I have read the `CONTRIBUTING.md` document
- [ ] I have signed the CA (see https://cla.pivotal.io/sign/rabbitmq)
- [x ] All tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in related repositories

## Further Comments

If this is a relatively large or complex change, kick off the discussion by
explaining why you chose the solution you did and what alternatives you
considered, etc.

### CODE REVIEW

File: projects/RabbitMQ.Client/client/impl/AutorecoveringConnection.cs [hunk 1]
```diff
@@ -896,6 +896,12 @@ private void EnsureIsOpen()
         private void HandleTopologyRecoveryException(TopologyRecoveryException e)
         {
             ESLog.Error("Topology recovery exception", e);
+            if (e.InnerException is AlreadyClosedException || e.InnerException is OperationInterruptedException || e.InnerException is TimeoutException)
+            {
+                throw e;
+            }
+            ESLog.Error($"Will not throw Topology Exception {e.InnerException?.GetType().FullName}.", e);
```
Comment (michaelklishin) [inline projects/RabbitMQ.Client/client/impl/AutorecoveringConnection.cs:903]: I assume this is mean to say "Will now throw". I also suggest a slightly different wording, something like

```
Will throw a topology recovery exception of type … to retry recovery
```

and perhaps it should be a warning and not an error?

**Review by michaelklishin — CHANGES_REQUESTED** [review]
The wording of the log message needs a bit of work.

File: projects/RabbitMQ.Client/client/impl/AutorecoveringConnection.cs (hunk 1 above)
Comment (pierresetteskog) [reply projects/RabbitMQ.Client/client/impl/AutorecoveringConnection.cs]: fixed in the last commit, my intention was to log why we have ignored the exception. And have something to search for if we want to add more exceptions in the future, I can also remove the logrow.

Comment (pierresetteskog) [reply projects/RabbitMQ.Client/client/impl/AutorecoveringConnection.cs]: let me know if i should make a pr to master also or instead,its exactly the same solution

**Review by michaelklishin — APPROVED** [review]

**Comment by michaelklishin** [comment]
@pierresetteskog if this is applicable to master, please submit another PR. This branch cannot be easily merged and I haven't looked at this part of the client in a while.

**Pull request merged**